vms/platformvm/state/stakers_crud_ops_test.go

```diff
@@ -0,0 +1,535 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/manager"
+	"github.com/ava-labs/avalanchego/database/versiondb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/version"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+const (
+	pending uint8 = iota + 1
+	current
+)
+
+// Check that CRUD operations over current stakers behaves uniformly for diffs and base state
+func TestCurrentValidatorsCRUD(t *testing.T) {
```

Comment:
I feel like it might make sense to consolidate these 4 tests into 2 -- one for "current" operations and one for "pending" operations, or one for validators and one for delegators

Reply:
Yes, there is some common code indeed, but I like the idea of testing a single thing, in isolation, even without validators in the genesis? This is similar to what we'd do in a DB. Let's see what other reviewers think of this code duplication here.